### ISSUE TITLE
Invalid URL when working with rest/admin/services

### ISSUE THREAD

**Issue by dpbayer** [Esri/arcgis-rest-js#329]
This is a really great library, that we are using heavily on a project.

I have encountered an issue with using the library to create hosted services in an on premise portal....

I can successfully create a service, using the createService method. When I try to addToDefinition, the code fails.
```
// This method succeeds
https://<on_premise_portal_url>/portal/sharing/rest/content/users/<username>/createService

// Calls to this URL fail
http://<on_premise_portal_url>/server/rest/admin/services/Hosted/<serviceName>/FeatureServer/addToDefinition
```
In the browser log, the code is accessing an invalid URL, to get the server info and hence the token, so that it can work out what token to use.

I have tracked down the issue, inside of UserSession.getTokenForServer

The code below, does not take into account, that the url could be /rest/admin/services  and so creates an invalid url
```javascript
private getTokenForServer(
    url: string,
    requestOptions?: ITokenRequestOptions
  ) {
    const [root] = url.split("/rest/services/");
    const existingToken = this.trustedServers[root];
```

To work around the issue, I have added the following to my code
```javascript
(UserSession.prototype as any).getTokenForServer = function(
    url: string,
    requestOptions?: ITokenRequestOptions
  )  {
      let spliter = "";
      if (url.indexOf("/rest/services/")>-1) {
        spliter = "/rest/services/";
      }
      else if (url.indexOf("/rest/admin/services/")>-1) {
        spliter = "/rest/admin/services/";
      }
    const [root] = url.split(spliter);
```

This issue was not showing itself with online, I think because the getToken method, is satisfied by the portal url.

**Comment by jgravois** [comment]
thanks for the detailed bug report @dpbayer!

**Comment by tomwayson** [comment]
@dpbayer the fix for this has been released in [v1.11.0](https://github.com/Esri/arcgis-rest-js/releases/tag/v1.11.0).
